opal/mca/memory/linux/memory_linux_component.c

```diff
@@ -100,6 +105,26 @@ opal_memory_linux_component_t mca_memory_linux_component = {
 static bool ptmalloc2_available = MEMORY_LINUX_PTMALLOC2;
 static bool ummunotify_available = MEMORY_LINUX_UMMUNOTIFY;
 
+#if MEMORY_LINUX_MALLOC_ALIGN_ENABLED
+
+static void *(*prev_malloc_hook)(size_t, const void *);
+
+/* This is a memory allocator hook. The purpose of this is to make
+ * every malloc aligned since this speeds up IB HCA work.
```

Comment:
This functionality is for general OS-bypass hardware, not specific to IB, right?

If it's for general OS-bypass hardware (which I think it is), the comment should probably state that (to avoid "What is IB-specific functionality doing down here?" reactions).